paasta_tools/flink_tools.py

```diff
@@ -361,3 +409,443 @@ def get_flink_overview_from_paasta_api_client(
+        - jobs: List of FlinkJobDetails
+    """
+    # Collect pod counts
+    pod_running_count = 0
+    pod_evicted_count = 0
+    pod_other_count = 0
+
+    for pod in status.get("pod_status", []):
+        if pod["phase"] == "Running":
+            pod_running_count += 1
+        elif pod["phase"] == "Failed" and pod.get("reason") == "Evicted":
+            pod_evicted_count += 1
+        else:
+            pod_other_count += 1
+
+    pods_total_count = pod_running_count + pod_evicted_count + pod_other_count
+
+    pod_counts: PodCounts = {
+        "running": pod_running_count,
+        "evicted": pod_evicted_count,
+        "other": pod_other_count,
+        "total": pods_total_count,
+    }
+
+    # Collect job counts if overview is available
+    job_counts: Optional[JobCounts] = None
+    taskmanagers = None
+    slots_available = None
+    slots_total = None
+
+    if overview:
+        jobs_total_count = (
+            overview.jobs_running
+            + overview.jobs_finished
+            + overview.jobs_failed
+            + overview.jobs_cancelled
+        )
+        job_counts = {
+            "running": overview.jobs_running,
+            "finished": overview.jobs_finished,
+            "failed": overview.jobs_failed,
+            "cancelled": overview.jobs_cancelled,
+            "total": jobs_total_count,
+        }
+        taskmanagers = overview.taskmanagers
+        slots_available = overview.slots_available
+        slots_total = overview.slots_total
+
+    return {
+        "state": status["state"],
+        "pod_counts": pod_counts,
+        "job_counts": job_counts,
+        "taskmanagers": taskmanagers,
+        "slots_available": slots_available,
+        "slots_total": slots_total,
+        "jobs": jobs,
+    }
+
+
+def format_flink_state_and_pods(
+    state: str,
+    pod_counts: PodCounts,
+    job_counts: Optional[JobCounts],
+    taskmanagers: Optional[int],
+    slots_available: Optional[int],
+    slots_total: Optional[int],
+) -> List[str]:
+    """Format state, pods, jobs summary, taskmanagers, and slots.
+
+    :param state: Flink cluster state (e.g., "running", "stopped")
+    :param pod_counts: Dict with running, evicted, other, total pod counts
+    :param job_counts: Dict with running, finished, failed, cancelled, total job counts (or None)
+    :param taskmanagers: Number of taskmanagers (or None)
+    :param slots_available: Number of available slots (or None)
+    :param slots_total: Total number of slots (or None)
+    :returns: List of formatted strings like:
+        State: Running
+        Pods: 3 running, 0 evicted, 0 other, 3 total
+        Jobs: 1 running, 0 finished, 0 failed, 0 cancelled, 1 total
+        1 taskmanagers, 0/1 slots available
+    """
+    output: List[str] = []
+
+    # Format state with color
+    color = PaastaColors.green if state == "running" else PaastaColors.yellow
+    output.append(f"    State: {color(state.title())}")
+
+    # Format pod counts
+    evicted_str = (
```

Comment:
i personally dislike having types in variable names and would name this `formatted_evictions` or something, but not a blocker)